hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -387,6 +403,9 @@ public DatanodeDeletedBlockTransactions getTransactions(
           }
         }
         purgeTransactions(txnsToBePurged);
+        for (DeletedBlocksTransaction trx : txnsToBePurged) {
+          transactionRetryCountMap.remove(trx.getTxID());
```

Comment:
In fact this might be a bug: for purgeTransaction I didn't `transactionToDNsCommitMap` is cleaned up properly.

Add the cleaning up of `transactionToDNsCommitMap` here.